src/com/spike/Expr.java

```diff
@@ -24,7 +24,7 @@
 
   static class Binary extends Expr {
     final Expr left;
-    final Token operator;
+    final Token operator; // NOTE: operator might have a form of op=
```

Comment:
Compound assignment expression should be decomposed as follows: `y -= 10` -> `y = y - 10`, so There is no need to store compound assignment operators in Binary expression. Because in this implementation, expression `x += 5` will be parsed as `Assign(x, Binary(x, '+=', 5))`, which is equivalent to `x = x += 5`. Check next comment on this.

Reply:
One way to resolve that issue would be to replace `Token` with `TokenType` but then we won't be able to report errors correctly at all (since `left` and `right` fields have no debug information)

Reply:
Then it would be better to create another expression for this like 
`Expr.CompoundAssign(Token name, Token operator, Expr value)`. In this case we will decouple the functionality. And `Expr.Binary` won't be overloaded with condition statements in interpreter + it will be easier to understand code.

Reply:
I just thought we can create tokens with new `TokenType` but old debug information, but that looks somewhat weird.

Reply:
What do you think about creating new class `Expr.CompoundAssign` which will handle this expression. Inside it will hold these tokens (`+=`, `-=` etc.), this should solve the problem, because we know the semantic of expression and will be able to evaluate it accordingly. And also no need to create new TokenType operators.